packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -183,11 +204,53 @@ export class HttpRoute extends Resource implements IHttpRoute {
       routeKey: props.routeKey.key,
       target: `integrations/${integration.integrationId}`,
       authorizerId: authBindResult?.authorizerId,
-      authorizationType: authBindResult?.authorizationType ?? 'NONE', // must be explicitly NONE (not undefined) for stack updates to work correctly
+      authorizationType: Lazy.string({
+        produce: () => {
+          if (this.iamEnabled) {
+            return 'AWS_IAM';
```

Comment:
Should `AWS_IAM` and `NONE` perhaps be enums?

Reply:
I don't mind leaving this as a hardcoded strings